src/main/java/org/baeldung/spring/CaptchaConfig.java

```diff
@@ -0,0 +1,22 @@
+package org.baeldung.spring;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
+
+@Configuration
+@ComponentScan(basePackages = { "org.baeldung.captcha" })
+@ConditionalOnProperty(prefix = "google.recaptcha.key", name = { "site", "secret" })
+public class CaptchaConfig {
+    @Bean
+    public ClientHttpRequestFactory clientHttpRequestFactory() {
```

Comment:
I wrote an [experiment](https://gist.github.com/anonymous/fe8d1c92336e179b2769734bd76b82fa) to compare to two.  The RestTemplate does look nicer.  My only concern is that someone copying the code might allocate RestTemplate inside the method; like I did at first.  I would like to put a footnote about that being upto 9x slower. In the commit I'm preparing; I switched to RestTemplate.

In the Article I tried to connect my dots to connect timeout / read timeout.  If you still fell extraneous about them; I'll cut it